src/sst/tests/test_actions.py

```diff
@@ -81,6 +84,20 @@ def protected_raiser():
 
         self.assertRaises(TestException, protected_raiser)
 
+    def test_retry_on_exception_fails_with_full_traceback(self):
+        max_retries = 1
+
+        @actions.retry_on_exception(TestException, retries=max_retries)
+        def protected_raiser():
+            return self.raise_exception(times=max_retries + 1)
+
+        try:
+            protected_raiser()
+        except:
```

Comment:
Need to check that exception was raised (only exception handler is executed after call to `protected_raiser`)

Reply:
That is checked with a test above `test_retry_on_exception_fails_with_max_retries`.